instrumentation/github.com/aws/aws-lambda-go/otellambda/lambda.go

```diff
@@ -0,0 +1,351 @@
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-lambda-go/otellambda"
+)
+
+var tp *sdktrace.TracerProvider
+var errorLogger = log.New(log.Writer(), "OTel Lambda Error: ", 0)
+
+func init() {
+	otel.SetTextMapPropagator(xray.Propagator{})
+}
+
+func initTracerProvider() {
+	ctx := context.Background()
+
+	exp, err := otlptracegrpc.New(ctx, otlptracegrpc.WithInsecure())
+	if err != nil {
+		errorLogger.Printf("failed to initialize exporter: %v\n", err)
+		return
+	}
+
+	detector := lambdadetector.NewResourceDetector()
+	res, err := detector.Detect(ctx)
+	if err != nil {
+		errorLogger.Printf("failed to detect lambda resources: %v\n", err)
+		return
+	}
+
+	tp = sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exp),
+		sdktrace.WithIDGenerator(xray.NewIDGenerator()),
+		sdktrace.WithResource(res),
+	)
+
+	// Set the traceprovider
+	otel.SetTracerProvider(tp)
+}
+
+func errorHandler(e error) func(context.Context, interface{}) (interface{}, error) {
+	return func(context.Context, interface{}) (interface{}, error) {
+		return nil, e
+	}
+}
+
+// Ensure handler takes 0-2 values, with context
+// as its first value if two arguments exist
+func validateArguments(handler reflect.Type) (bool, error) {
+	handlerTakesContext := false
+	if handler.NumIn() > 2 {
+		return false, fmt.Errorf("handlers may not take more than two arguments, but handler takes %d", handler.NumIn())
+	} else if handler.NumIn() > 0 {
+		contextType := reflect.TypeOf((*context.Context)(nil)).Elem()
+		argumentType := handler.In(0)
+		handlerTakesContext = argumentType.Implements(contextType)
+		if handler.NumIn() > 1 && !handlerTakesContext {
+			return false, fmt.Errorf("handler takes two arguments, but the first is not Context. got %s", argumentType.Kind())
+		}
+	}
+
+	return handlerTakesContext, nil
+}
+
+// Ensure handler returns 0-2 values, with an error
+// as its first value if any exist
+func validateReturns(handler reflect.Type) error {
+	errorType := reflect.TypeOf((*error)(nil)).Elem()
+
+	switch n := handler.NumOut(); {
+	case n > 2:
+		return fmt.Errorf("handler may not return more than two values")
+	case n > 1:
+		if !handler.Out(1).Implements(errorType) {
+			return fmt.Errorf("handler returns two values, but the second does not implement error")
+		}
+	case n == 1:
+		if !handler.Out(0).Implements(errorType) {
+			return fmt.Errorf("handler returns a single value, but it does not implement error")
+		}
+	}
+
+	return nil
+}
+
+// Wraps and calls customer lambda handler then unpacks response as necessary
+func wrapperInternals(handlerFunc interface{}, event reflect.Value, ctx context.Context, takesContext bool) (interface{}, error) {
```

Comment:
```suggestion
func wrapperInternals(ctx context.Context, handlerFunc interface{}, event reflect.Value, takesContext bool) (interface{}, error) {
```

The context should always be the first argument.